repository/src/main/java/com/epam/esm/repository/impl/CertificateRepositoryImpl.java

```diff
@@ -118,4 +119,24 @@ public Long count() {
                 .getSingleResult();
     }
 
+    private <T> void prepareSearchQuery(AbstractQuery<T> query, Root<Certificate> root, List<String> tagNames, String name) {
+        CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+        List<Predicate> predicates = new ArrayList<>();
+        if (!StringUtils.isBlank(name)) {
+            Predicate predicateForName = criteriaBuilder.like(root.get("name"), "%" + name + "%");
+            predicates.add(predicateForName);
+        }
+        if (tagNames != null && tagNames.size() != 0) {
```

Comment:
Please, look into the CollectionUtils#isNotEmpty

Reply:
Done